### ISSUE TITLE
Problems with script execution

### ISSUE THREAD

**Issue by wzygxlz** [dnGrep/dnGrep#1219]
Problem description: If a script with an incorrect expression is run before, none of the subsequent scripts will run correctly.
Reproduce the steps:

1. Create a new script called 0.gsc with the following contents:

```
set folder z:\2-Tools\Tools\vsCode.x64 
set searchtype Regex
set searchfor [a-z]((%3A|:)(\/|\\{1,2})2-Tools\3Tools\3vsCode\.x64))
search
```
where the search condition has an extra parenthesis and is the wrong search expression.

2. Create a new script called 1.gsc, which is similar to 0.gsc, but with the error corrected, as follows:
```
set folder z:\2-Tools\Tools\vsCode.x64
set searchtype Regex
set searchfor [a-z]((%3A|:)(\/|\\{1,2})2-Tools\3Tools\3vsCode\.x64)
search
```

3. Run 0.gsc in the Scripts menu first and you will be prompted: 
Regular expression is not valid!
script complete.

4. Select Run 1.gsc in the Scripts menu and you will be prompted: Search canceled in 0.5s,That means the search was automatically terminated.

**Comment by doug24** [comment]
Yes, this is a bug - missed resetting a flag after the first script ended in an error.

**Comment by doug24** [comment]
Fixed in [4.2.113](https://github.com/dnGrep/dnGrep/releases/tag/v4.2.113.0)
